### ISSUE TITLE
Code duplication

### ISSUE THREAD

**Issue by sbfnk** [epiforecasts/covid19-forecast-hub-europe#469]
There is duplicate code and text in the ensemble/evaluation reports and the evaluation code:
https://github.com/epiforecasts/covid19-forecast-hub-europe/tree/main/code/reports
https://github.com/epiforecasts/covid19-forecast-hub-europe/tree/main/code/evaluation
Ideally these should share a common code base one way or the other.

**Comment by sbfnk** [comment]
With further multiplication here:
https://github.com/epiforecasts/europe-covid-forecast/tree/master/reports

**Comment by kathsherratt** [comment]
Relatedly - I was also wondering if there is a neater way of organising the parts that outside of the "core" of the hub. I think of the core as the shared structure with the other hubs: `data-processed`, `data-truth`, `data-locations`, i.e. the structure required for `covidHubUtils` to work. At the moment everything else is a bit of a mixed bag inherited from the DE/PL and US hubs. 

For example I think the code for evaluation, report, and [ensemble](https://github.com/epiforecasts/covid19-forecast-hub-europe/tree/main/code/ensemble) are now all relatively stand-alone. We could do the same as with the [validation](https://github.com/epiforecasts/covid19-forecast-hub-europe-validations) code, and just pull them out into new repos (potentially included in this repo as sub-modules), which might make it a little clearer / easier to work on them independently.

I guess some of this might then start to overlap with [`covidHubUtils`](https://github.com/reichlab/covidHubUtils) and [`covidEnsembles`](https://github.com/reichlab/covidEnsembles) packages though (if it doesn't already).

**Comment by sbfnk** [comment]
Yes, these are all good ideas, and I think ultimately we should aim to help generalise `covidHubUtils` and `covidEnsembles` (and whatever package we might want to add, e.g. for generating the reports - see also ongoing discussion at https://github.com/epiforecasts/europe-covid-forecast/pull/52) to a degree that we have minimal code that's specific to this repo.
